server/webapp/WEB-INF/rails.new/app/controllers/api_v3/admin/environments_controller.rb

```diff
@@ -0,0 +1,129 @@
+##########################################################################
+# Copyright 2017 ThoughtWorks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+module ApiV3
+  module Admin
+    class EnvironmentsController < BaseController
+      before_action :check_admin_user_and_401
+      before_action :load_local_environment, only: [:put, :patch, :destroy]
```

Comment:
But according to your json, if an environment's origin ends up being `remote` or `config_repo` instead of `merged`(ie specified partially in local and in remote), then it should show an error?